File: source/examples/CRUD/Create.java
// Open a thread-safe transaction.
realm.executeTransaction(r -> {
    // Instantiate the class using the factory function.
    Dog dog = realm.createObject(Dog.class);

    // Configure the instance.
    dog.setName("Max");
    dog.setAge(5);

    // Create a Person with a primary key.
    int primaryKeyValue = 123;
    Person person = realm.createObject(Person.class, primaryKeyValue);
});

File: source/examples/CRUD/Create.kt
// Open a thread-safe transaction.
realm.executeTransaction { realm ->
    // Instantiate the class using the factory function.
    val dog = realm.createObject<Dog>()

    // Configure the instance.
    dog.name = "Max"
    dog.age = 5

    // Create a Person with a primary key.
    val primaryKeyValue = 123
    val person = realm.createObject<Person>(primaryKeyValue)
}
